src/components/Atoms/SocialIcons/Icon/Icon.js

```diff
@@ -24,29 +80,41 @@ const HelperText = styled.span`
 `;
 
 const Icon = ({
-  href, target, icon, brand, title, ...restProps
+  href, target, icon, brand, title, isHeader, id, ...restProps
 }) => (
   <StyledLink
     href={href}
     target={`_${target}`}
     {...restProps}
     title={title}
     rel="noopener noreferrer"
-    data-test="header-esu-icon"
+    data-testid={`${isHeader ? 'header' : 'icon'}-${kebabCase(id)}`}
```

Comment:
Only passing this in the new Header2024, so the pre-existing Header Classic is not affected